style/js/shield_defs.js

```diff
@@ -54,6 +55,17 @@ function roundedRectShield(
   };
 }
 
+function banneredShield(baseDef, modifiers) {
+  let newShield = Util.cp(baseDef);
+  newShield.modifiers = modifiers;
+
+  //Restore clobbered pointers
+  newShield.backgroundImage = baseDef.backgroundImage;
+  newShield.textLayoutConstraint = baseDef.textLayoutConstraint;
+
+  return newShield;
+}
+
```

Comment:
I think a shallow clone is all that's needed here.  i.e:
```
function banneredShield(baseDef, modifiers) {
  return {
    ...baseDef,
    modifiers: modifiers,
  };
}
```

Reply:
Gonna try this and see if it still works.  I like the sound of the shallow clone better than some mysterious clobber comments that nobody will understand.

Reply:
Yup... this is the way.

Ready for reapproval.